### PR TITLE
[Downloads 3/8] Mount directories read-only where possible

### PR DESCRIPTION
**Pull request by natano**

### CODE REVIEW

File: src/PrefsManager.cpp [hunk 1]
```diff
@@ -500,6 +510,16 @@ void PrefsManager::ReadDefaultsFromIni( const IniFile &ini, const RString &sSect
 	IPreference::ReadAllDefaultsFromNode( ini.GetChild(sSection) );
 }
 
+void PrefsManager::TranslateDeprecatedFlags()
+{
+	if(m_sAdditionalFolders.Get() != "")
```
Comment (teejusb) [inline src/PrefsManager.cpp]: `.empty()` preferred

File: src/PrefsManager.h [hunk 2]
```diff
@@ -352,6 +355,11 @@ class PrefsManager
 protected:
 	void ReadPrefsFromFile( const RString &sIni, const RString &sSection, bool bIsStatic );
 	void ReadDefaultsFromFile( const RString &sIni, const RString &sSection );
+	void TranslateDeprecatedFlags();
+
+	Preference<RString>	m_sAdditionalSongFolders;	// compat
```
Comment (teejusb) [inline src/PrefsManager.h]: // deprecated

File: src/StepMania.cpp [hunk 3]
```diff
@@ -917,19 +917,30 @@ static void MountTreeOfZips( const RString &dir )
 		GetDirListing( path + "/*.zip", zips, false, true );
 		GetDirListing( path + "/*.smzip", zips, false, true );
 
-		for( unsigned i = 0; i < zips.size(); ++i )
+		for (const auto& zip : zips)
```
Comment (teejusb) [inline src/StepMania.cpp:920]: viva la c++11

File: src/StepMania.cpp [hunk 4]
```diff
@@ -917,19 +917,30 @@ static void MountTreeOfZips( const RString &dir )
...
+			FILEMAN->Mount( "zip", zip, "/" );
 		}
 
 		GetDirListing( path + "/*", dirs, true, true );
 	}
 }
 
+static void MountFolders(const RString &type, const RString &realPathList, const RString &mountPoint)
+{
+	if (realPathList != "")
```
Comment (teejusb) [inline src/StepMania.cpp]: `.empty()`

File: src/PrefsManager.cpp (hunk 1 above)
Comment (natano) [reply src/PrefsManager.cpp]: changed

File: src/PrefsManager.h (hunk 2 above)
Comment (natano) [reply src/PrefsManager.h]: fixed

File: src/StepMania.cpp (hunk 3 above)
Comment (natano) [reply src/StepMania.cpp:920]: C++11 rocks!

File: src/StepMania.cpp (hunk 4 above)
Comment (natano) [reply src/StepMania.cpp]: fixed

**Pull request merged**